File: waiter-portal/src/types/order.ts
export interface OrderLine {
  id: string;
  orderId: string;
  inventoryItemId: string;
  itemName: string;
  unitPrice: number;
  quantity: number;
  notes: string | null;
  varietySelection: string | null;
}

export interface Order {
  id: string;
  orderNumber: string;
  retailerId: string;
  waiterName: string | null;
  tableNumber: string | null;
  customerNotes: string | null;
  customerPhone: string | null;
  status: string;
  totalAmount: number;
  displayCurrency: string;
  lines: OrderLine[];
  createdAt: string;
  updatedAt: string;
}

export interface CreateOrderLineRequest {
  inventoryItemId: string;
  itemName: string;
  unitPrice: number;
  quantity: number;
  notes?: string;
  varietySelection?: string;
}

export interface CreateOrderRequest {
  retailerId: string;
  waiterName?: string;
  tableNumber?: string;
  customerNotes?: string;
  customerPhone?: string;
  displayCurrency?: string;
  lines: CreateOrderLineRequest[];
}

export interface CreateOrderResponse {
  id: string;
  orderNumber: string;
}

export type OrderStatus =
  | "Pending"
  | "Confirmed"
  | "Preparing"
  | "Ready"
  | "Served"
  | "Cancelled";

File: src/Inventory.API/Data/Entities/Order.cs
namespace Inventory.API.Data.Entities;

public sealed class Order
{
    public Guid Id { get; set; }
    public string OrderNumber { get; set; } = "";
    public string RetailerId { get; set; } = "";
    public string? WaiterName { get; set; }
    public string? TableNumber { get; set; }
    public string? CustomerNotes { get; set; }
    public string? CustomerPhone { get; set; }
    public string Status { get; set; } = "Pending";
    public decimal TotalAmount { get; set; }
    public string DisplayCurrency { get; set; } = "GHS";
    public List<OrderLine> Lines { get; set; } = new();
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
}
